nemo/lightning/data.py

```diff
@@ -172,18 +174,15 @@ def add_megatron_sampler(
     Returns:
         DataLoader: A new DataLoader instance with the configured Megatron sampler.
     """
-
-    from megatron.core import parallel_state
-
     if dataloader_type == 'single':
         batch_sampler = MegatronPretrainingSampler(
             total_samples=len(dataloader.dataset),
             consumed_samples=consumed_samples,
             micro_batch_size=micro_batch_size,
             global_batch_size=global_batch_size,
             rampup_batch_size=rampup_batch_size,
-            data_parallel_rank=parallel_state.get_data_parallel_rank(),
-            data_parallel_size=parallel_state.get_data_parallel_world_size(),
+            data_parallel_rank=rank,
+            data_parallel_size=world_size,
```

Comment:
Why do we need these changes? It feels risky to pass in rank and world_size as args instead of infer from parallel_state?

Reply:
Can we initialize them inside this method instead of making them an arg?

Reply:
I want to be able to use this from the HF model and that does not use parallel_state, so I pass them as params.

Reply:
In that case, can you use https://pytorch.org/docs/stable/distributed.html#torch.distributed.get_rank for HF model?

Reply:
for mcore, I'm going to [use](https://github.com/NVIDIA/NeMo/pull/10962/files#diff-c423cc7981d5621754bb3e5e509fceaa9adb8537b269b7def042995cfa21d529R74) `parallel_state` since it already has that information and for hf I'm going to use torch.distributed.get_rank/get_world_size as in hf_dataset.py